### DISCUSSION TITLE
[Bug] the queue's messages, span too long, so do flow control

### DISCUSSION THREAD

**Discussion by lfs1985** [apache/rocketmq#7133] [General]
There are many warn logs in the production environment。
![image](https://github.com/apache/rocketmq/assets/44808351/16803beb-7565-4b54-91ae-6a7a00773daf)
The message has been processed，but the client's (in processQueue)minOffset no changed for long time。 
Restart the client, the problem can be solved, but many messages will be repeated consumpted。
We try to reproduce it，but have no result。

rmq client: 4.7.1
rmq broker: 4.7.1
os: centos 7

**Comment by HScarb** [comment]
消费处理队列 ProcessQueue 中的数据堆积过多，超过阈值（默认为2000）。
消息拉取速度大于消费速度导致。

**Comment by cserwen** [comment]
It should be caused by this problem  https://github.com/apache/rocketmq/pull/3861

As for the reason for throwing an exception, it may be that `PROPERTY_CONSUME_START_TIMESTAMP` has not been set when cleaning up the expired message, so an exception is thrown at here.
https://github.com/apache/rocketmq/blob/9f95a972e10e0681bc3f2d00e9957aa212e897b5/client/src/main/java/org/apache/rocketmq/client/impl/consumer/ProcessQueue.java#L88C1-L89C1
